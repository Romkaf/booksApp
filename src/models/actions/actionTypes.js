export const FETCH_BOOKS = 'FETCH_BOOKS';
export const FETCH_BOOKS_SUCCESS = 'FETCH_BOOKS_SUCCESS';
export const FETCH_BOOKS_ERROR = 'FETCH_BOOKS_ERROR';
export const SET_AUTHOR_FILTER = 'SET_AUTHOR_FILTER';
export const SET_TITLE_FILTER = 'SET_TITLE_FILTER';
export const SET_CURRENT_PAGE = 'SET_CURRENT_PAGE';
