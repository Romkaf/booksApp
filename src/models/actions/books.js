import {
	FETCH_BOOKS,
	FETCH_BOOKS_SUCCESS,
	FETCH_BOOKS_ERROR,
} from './actionTypes';

import { fetchGetBooks } from '@api';

const fetchBooksRequest = () => ({
	type: FETCH_BOOKS,
});

const fetchBooksSuccess = (data) => ({
	type: FETCH_BOOKS_SUCCESS,
	payload: data,
});

const fetchBooksError = (err) => ({
	type: FETCH_BOOKS_ERROR,
	payload: err,
});

export const fetchBooks = (param) => async (dispatch) => {
	try {
		const { titleFilter, authorFilter } = param;
		dispatch(fetchBooksRequest());
		const res = await fetchGetBooks(titleFilter, authorFilter);
		dispatch(fetchBooksSuccess(res.data));
	} catch (err) {
		dispatch(fetchBooksError(err.message));
	}
};
