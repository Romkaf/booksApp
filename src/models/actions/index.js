import {
	FETCH_BOOKS,
	FETCH_BOOKS_SUCCESS,
	FETCH_BOOKS_ERROR,
} from './actionTypes';

import { fetchGetBooks } from '@api';

const fetchBooksRequest = () => ({
	type: FETCH_BOOKS,
});

const fetchBooksSuccess = (data) => ({
	type: FETCH_BOOKS_SUCCESS,
	payload: data,
});

const fetchBooksError = (err) => ({
	type: FETCH_BOOKS_ERROR,
	payload: err,
});

export const fetchBooks = () => async (dispatch) => {
	try {
		dispatch(fetchBooksRequest());
		const res = await fetchGetBooks();
		dispatch(fetchBooksSuccess(res.data));
	} catch (err) {
		dispatch(fetchBooksError(err.message));
	}
};
