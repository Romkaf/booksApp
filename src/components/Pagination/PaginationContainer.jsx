import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { createPages } from '@utils/createPages';
import { setCurrentPage } from '@models/actions';
import Pagination from './Pagination';

const PaginationContainer = () => {
	const dispatch = useDispatch();

	const { currentPage, perPage } = useSelector((state) => state.pagination);
	const totalCount = useSelector((state) => state.books.totalCount);

	const handlePageClick = (page) => () => dispatch(setCurrentPage(page));

	const pagesCount = Math.ceil(totalCount / perPage);
	const pages = createPages(pagesCount, currentPage);

	return (
		<Pagination
			pages={pages}
			onPageClick={handlePageClick}
			currentPage={currentPage}
		/>
	);
};

export default PaginationContainer;
