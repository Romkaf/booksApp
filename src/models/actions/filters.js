import { SET_TITLE_FILTER, SET_AUTHOR_FILTER } from './actionTypes';

export const setAuthorFilter = (value) => ({
	type: SET_AUTHOR_FILTER,
	payload: value,
});

export const setTitleFilter = (value) => ({
	type: SET_TITLE_FILTER,
	payload: value,
});
