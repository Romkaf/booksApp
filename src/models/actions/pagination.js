import { SET_CURRENT_PAGE } from './actionTypes';

export const setCurrentPage = (number) => ({
	type: SET_CURRENT_PAGE,
	payload: number,
});
