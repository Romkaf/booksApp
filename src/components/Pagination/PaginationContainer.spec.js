import React from 'react';
import PaginationContainer from './PaginationContainer';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { setCurrentPage } from '@models/actions';

describe('PaginationContainer', () => {
	const mockStore = configureMockStore([thunk]);

	const store = mockStore({
		books: { books: [], totalCount: 10 },
		pagination: { currentPage: 1, perPage: 5 },
	});

	const wrapper = mount(
		<Provider store={store}>
			<PaginationContainer />
		</Provider>,
	);

	it('should render component', () => {
		const component = wrapper.find('PaginationContainer');
		expect(component).toMatchSnapshot();
	});

	it('should handler return action', () => {
		const handler = wrapper.find('Pagination').props().onPageClick;

		expect(handler(1)()).toStrictEqual(setCurrentPage(1));
	});
});
