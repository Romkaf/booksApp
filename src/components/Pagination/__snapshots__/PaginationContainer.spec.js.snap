// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationContainer should render component 1`] = `
<PaginationContainer>
  <Pagination
    currentPage={1}
    onPageClick={[Function]}
    pages={
      Array [
        1,
        2,
      ]
    }
  >
    <ul
      className="root"
    >
      <li
        key="0"
      >
        <button
          aria-label="pagination-page"
          className="page pageCurrent"
          onClick={[Function]}
          tabIndex={-1}
        >
          1
        </button>
      </li>
      <li
        key="1"
      >
        <button
          aria-label="pagination-page"
          className="page"
          onClick={[Function]}
          tabIndex={0}
        >
          2
        </button>
      </li>
    </ul>
  </Pagination>
</PaginationContainer>
`;
